build.py

```
@@ -3,45 +3,177 @@
+                    match = self.import_re.match(line.strip())
+                    if match:
+                        imports.add(match.group(1))
+        except Exception:
+            pass
+        return imports
+
+    def get_installed_packages(self) -> set[str]:
+        """Queries the current environment for installed package names."""
+        try:
+            res = subprocess.run(
+                ["uv", "pip", "list"],
```

Ruff (0.15.12)

[error] Starting a process with a partial executable path

(S607)